web/postgres_typemap.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+func toSpannerTypePostgres(srcType string, spType string, mods []int64) (ddl.Type, []internal.SchemaIssue) {
+
+	switch srcType {
+	case "bool", "boolean":
+		switch spType {
+		case ddl.String:
+			return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, []internal.SchemaIssue{internal.Widened}
+		case ddl.Int64:
+			return ddl.Type{Name: ddl.Int64}, []internal.SchemaIssue{internal.Widened}
+		default:
+			return ddl.Type{Name: ddl.Bool}, nil
+		}
+
+	case "bigserial":
+		switch spType {
+		case ddl.String:
+			return ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, []internal.SchemaIssue{internal.Widened}
```

Comment:
I think issues should include internal.Serial in addition to widening.

Reply:
Yeah, added that issue for serial and big serial.

One doubt: 
For issues of Numeric and Decimal, the main reason we are using that with FLOAT64 is because it lacks precision. but in case of converting it to STRING, i think the only issue should be Widened, right?
Similar thing for Timestamp and Datetime issue, as those are more close to timestampz in postgres and timestamp in mysql respectively. Should we have those as issue when using STRING type?

Reply:
For Numeric and Decimal: yes, that's correct.

For Timestamp and Datetime: yes, generating Widening when we translate to STRING makes sense.

Spanner now has NUMERIC support, but we haven't added it to HarbourBridge yet. I'd been waiting till the new functionality we're working on now had landed (I didn't want to make changes in code that was already being changed). However, given the structure of this PR, that's not really an issue (the relevant code is being duplicated and the toddl.go files are not being changed).

So I'm wondering if we should just add NUMERIC support now. I'll draft some changes below.

Reply:
NUMERIC support now added.
Issue resolved here